xbmc/music/windows/GUIWindowVisualisation.cpp

```diff
@@ -240,3 +241,29 @@ void CGUIWindowVisualisation::FrameMove()
   }
   CGUIWindow::FrameMove();
 }
+
+void CGUIWindowVisualisation::ToggleOSD()
+{
+  CGUIDialogMusicOSD *pOSD = static_cast<CGUIDialogMusicOSD *>(g_windowManager.GetWindow(WINDOW_DIALOG_MUSIC_OSD));
+  if (pOSD)
+  {
+    if (pOSD->IsDialogRunning())
+    {
+      pOSD->Close();
+    }
+    else if (g_infoManager.GetShowInfo() &&
+             CSettings::Get().GetBool("pvrplayback.confirmchannelswitch") &&
+             g_application.CurrentFileItem().HasPVRChannelInfoTag())
+    {
+      // If only channel info but no player controls are currently visible
+      // and setting "confirm channel switch with OK" is active, do a channel
+      // switch instead of turning on the the player controls.
+      g_application.m_pPlayer->SwitchChannel(g_application.CurrentFileItem().GetPVRChannelInfoTag());
```

Comment:
@xhaggi: Does not work. And now I remember... Had it in the first place almost like you suggested and then found out that when hitting OK on my remote no ACTION_SELECT_ITEM, but instead ACTION_SHOW_OSD was created. Reason is here:
https://github.com/xbmc/xbmc/blob/master/system/keymaps/remote.xml#L196 and here https://github.com/xbmc/xbmc/blob/master/system/keymaps/remote.xml#L242
When changing the lines in question to <code>\<select\>Select\</select\></code> it works just fine with ACTION_SELECT_ITEM. So, the question is whether it is okay to change remote.xml that way? I'd say yes.